### PR TITLE
rsx-capture: unbreak

### PR DESCRIPTION
**Pull request by elad335**
fix rsx thread capture booting.

### CODE REVIEW

File: rpcs3/Emu/System.cpp [hunk 1]
```diff
@@ -486,7 +486,13 @@ bool Emulator::BootRsxCapture(const std::string& path)
 	GetCallbacks().on_run();
 	m_state = system_state::running;
 
-	fxm::make<rsx::rsx_replay_thread>(std::move(frame));
+	auto thread = fxm::make<rsx::rsx_replay_thread>(std::move(frame));
```
Comment (Nekotekina) [inline rpcs3/Emu/System.cpp]:
```suggestion
	auto thread = fxm::make<named_thread<rsx::rsx_replay_thread>>("RSX Replay", std::move(frame));
```

File: Utilities/Thread.cpp [hunk 2]
```diff
@@ -1404,7 +1404,7 @@ static LONG exception_handler(PEXCEPTION_POINTERS pExp)
 
 	if (pExp->ExceptionRecord->ExceptionCode == EXCEPTION_ACCESS_VIOLATION && addr64 < 0x100000000ull)
 	{
-		if (thread_ctrl::get_current() && handle_access_violation((u32)addr64, is_writing, pExp->ContextRecord))
+		if (handle_access_violation((u32)addr64, is_writing, pExp->ContextRecord))
```
Comment (Nekotekina) [inline Utilities/Thread.cpp]: No. Don't.

Comment (elad335) [reply Utilities/Thread.cpp]: This allows rsx mem faults on threads that don't have a thread_ctrl such as the rsx debugger and rsx replay. The other solution is to "clean" dirty rsx mem sections before touching it. @ruipin, what do you think?

Comment (Nekotekina) [reply Utilities/Thread.cpp]: Neither is a solution, debugger shouldn't trigger page faults at all.

Comment (Nekotekina) [reply Utilities/Thread.cpp]: Fixing debuggers is a lot of work anyway, let's merge only rsx-capture fix in this PR.

**Pull request merged**